packages/backend-defaults/src/entrypoints/urlReader/lib/ConfluenceUrlReader.ts

```diff
@@ -0,0 +1,273 @@
+  }
+
+  async loadConfluencePage(
+    pageId: string,
+    path: string,
+  ): Promise<FromReadableArrayOptions> {
+    const docItems: FromReadableArrayOptions = [];
+
+    try {
+      // fetch page
+      const { page, title, lastModifiedAt } = await this.fetchConfluencePage(
+        pageId,
+      );
+
+      // fetch its attachments
+      const attachments: Attachment[] =
+        await this.fetchConfluencePageAttachments(pageId);
+      for (const attachment of attachments) {
+        const imageData: any = await this.fetchAttachment(attachment);
+        docItems.push({
+          data: Readable.fromWeb(imageData),
+          path: `${path}/attachments/${attachment.title.replaceAll(' ', '-')}`,
+          lastModifiedAt: new Date(attachment.version.createdAt),
+        });
+      }
+
+      const pageMarkdown = this.replaceAttachmentLinks(attachments, page);
+      docItems.push({
+        data: Readable.from(pageMarkdown),
+        path: `${path}/${title}.md`,
+        lastModifiedAt: new Date(lastModifiedAt),
+      });
+
+      // fetch child pages and repeat
+      const children = await this.fetchChildPages(pageId);
+      for (const childPage of children) {
+        const childPageContent = await this.loadConfluencePage(
+          childPage.id,
+          `${path}/${title.toLocaleLowerCase().replaceAll(' ', '-')}`,
+        );
+        docItems.push(...childPageContent);
+      }
+    } catch (error) {
+      this.logger.error(
+        `Error reading page ${pageId} from confluence: ${error.message}`,
+      );
+    }
+
+    return docItems;
+  }
+
+  private replaceAttachmentLinks(attachments: Attachment[], markdown: string) {
+    let markdownToPublish = markdown;
+    for (const attachment of attachments) {
+      const attachmentTitle = attachment.title.replace(/ /g, '-');
+      const attachmentTitleInUrl = attachment.title.replace(/ /g, '%20');
+      // If .pdf, this regex matches Markdown links in the format [link text](url)
+      // If not ![alt text](url)
+      const regex = attachmentTitleInUrl.includes('.pdf')
+        ? new RegExp(`(\\[.*?\\]\\()(.*?${attachmentTitleInUrl}.*?)(\\))`, 'gi')
+        : new RegExp(
+            `(\\!\\[.*?\\]\\()(.*?${attachmentTitleInUrl}.*?)(\\))`,
+            'gi',
+          );
+      markdownToPublish = markdownToPublish.replace(
+        regex,
+        `$1./attachments/${attachmentTitle}$3`,
+      );
+    }
+    return markdownToPublish;
+  }
+
+  private async fetchChildPages(pageId: string): Promise<ChildPage[]> {
+    const { host, apiToken } = this.config;
+
+    const url = `https://${host}/wiki/api/v2/pages/${pageId}/children`;
+    const init = { headers: { Authorization: apiToken } };
+
+    const response = await this.fetchResponse(url, init);
+    const apiResponse = await response.json();
+
+    const childPages: ChildPage[] = apiResponse.results;
+    return childPages;
+  }
+
+  private htmlToMardDown(html: string) {
```

Comment:
typo: MardDown => Markdown

Reply:
Sure, will fix it. 
Let me know if there's anything else you'd like me to address. I can include it in this fix before committing

Reply:
Fixed it in 8d35b97